storage/innobase/log/log0recv.cc

```diff
@@ -3845,9 +3863,52 @@ byte *recv_dblwr_t::find_page(const page_id_t page_id,
   return result;
 }
 
+uint32_t recv_dblwr_t::find_first_page(const char* name, pfs_os_file_t file)
+{
+  uint32_t space= 0;
+  os_offset_t file_size= os_file_get_size(file);
+  if (file_size == (os_offset_t) -1) return space;
+  for (const page_t *page : pages)
+  {
+    uint32_t space_id= page_get_space_id(page);
+    if (page_get_page_no(page) > 0 || space_id == 0)
+next_page:
+      continue;
+    uint32_t flags= mach_read_from_4(
+      FSP_HEADER_OFFSET + FSP_SPACE_FLAGS + page);
+    page_id_t page_id(space_id, 0);
+    size_t page_size= fil_space_t::physical_size(flags);
+    if (file_size < 4 * page_size) goto next_page;
+    byte *read_page= static_cast<byte*>(
+      aligned_malloc(3 * page_size, page_size));
```

Comment:
Formatting: break the line between the `if` condition and the body, unless it is part of a very short block.

Also, do not use the legacy InnoDB style line breaks after `(`. Break lines after `,` or similar, but never after `(`, to be compatible with the default settings of the GNU Emacs `cc-mode`.